src/pages/CoursesPage.jsx

```diff
@@ -1,10 +1,31 @@
+import { useEffect, useState } from 'react';
 import { Gallery } from '../components/CoursesGallery/Gallery';
+import { fetchAllAdverts } from '../services';
 
 
 const CoursesPage = () => {
+    const [adverts, setAdverts] = useState([]);    
+
+    useEffect(() => {
+        async function getAdverts() {
```

Comment:
Move this function out of the useEffect please

Reply:
has been fixed